### PR TITLE
Overall improvement of post page looks

### PR DESCRIPTION
**Pull request by a1exymoroz**
### What is this PR about?
- Overall improvement of post page looks

### CODE REVIEW

File: src/app/components/author/author.component.html [hunk 1]
```diff
@@ -1,5 +1,5 @@
 @if (author) {
-  <div
+  <a
```
Comment (robinsonzimmermann) [inline src/app/components/author/author.component.html]: A better approach would be handling the link in the parent component. We can remove anything related to the profile URL and keep it as a `div`. Then, when this needs to be a link, the smart component can wrap it in a link tag.

File: src/app/components/post-item/post-item.component.html [hunk 2]
```diff
@@ -27,7 +27,12 @@
     </p>
     <div class="item-card__authors">
       @for (author of post.authors; track $index) {
-        <blog-author [author]="author" [muted]="true"></blog-author>
+        <a
+          [attr.data-role]="$any(author).url ? 'button' : ''"
+          [routerLink]="'/' + $any(author).url"
+          (click)="$event.stopPropagation()">
+          <blog-author [author]="author" [muted]="true"></blog-author>
+        </a>
```
Comment (robinsonzimmermann) [inline src/app/components/post-item/post-item.component.html]: The whole card is a link to the post, so we don't need a link to the author here. Please keep `blog-author` component only, without the link tag

File: src/app/features/post/post.component.html [hunk 3]
```diff
@@ -21,7 +21,12 @@ <h2>{{ post.excerpt }}</h2>
         </span>
         <div class="post__header-authors">
           @for (author of post.authors; track $index) {
-            <blog-author [author]="author" size="lg"></blog-author>
+            <a
+              [attr.data-role]="$any(author).url ? 'button' : ''"
```
Comment (robinsonzimmermann) [inline src/app/features/post/post.component.html]: It's already a link, so there's no need for a `data-role`
```suggestion
```

File: src/app/features/post/post.component.html [hunk 4]
```diff
@@ -21,7 +21,12 @@ <h2>{{ post.excerpt }}</h2>
...
+              [routerLink]="'/' + $any(author).url"
```
Comment (robinsonzimmermann) [inline src/app/features/post/post.component.html]: Do we need this `$any`?
```suggestion
              [routerLink]="'/' + author?.url"
```

File: src/app/features/post/post.component.html [hunk 5]
```diff
@@ -21,7 +21,12 @@ <h2>{{ post.excerpt }}</h2>
...
+              (click)="$event.stopPropagation()">
```
Comment (robinsonzimmermann) [inline src/app/features/post/post.component.html]: Can you please double-check whether this `stopPropagation` is needed? I actually can't see the use of it here
```suggestion
              >
```

File: src/app/features/post/post.component.html (hunk 3 above)
Comment (a1exymoroz) [reply src/app/features/post/post.component.html]: Done

File: src/app/features/post/post.component.html (hunk 5 above)
Comment (a1exymoroz) [reply src/app/features/post/post.component.html]: You are right. no need. I needed it for `post-item.component.html`

File: src/app/features/post/post.component.html (hunk 4 above)
Comment (a1exymoroz) [reply src/app/features/post/post.component.html]: I need it. I have an error
image
image

Comment (a1exymoroz) [reply src/app/features/post/post.component.html]: I can create some pipe or method instead of `$any`.

**Review by robinsonzimmermann — APPROVED** [review]

**Pull request merged**